### DISCUSSION TITLE
Importing Firebase throws error

### DISCUSSION THREAD

**Discussion by CynthiaBlue** [quasarframework/quasar#10504] [General - Components / Directives / etc]
I'm following one of Danny Connell's tutorials (he has some really good ones) about getting Quasar working with firebase.
This is the video tutorial:
https://youtu.be/la-0ulfn0_M

My little test project keeps throwing an error, however.
I've installed firebase and created the boot file.

This is the content of my boot file:

`import firebase from "firebase/app"
import "firebase/firestore"

const firebaseConfig = {
  apiKey: "stuff",
  authDomain: "stuff",
  projectId: "stuff",
  storageBucket: "stuff",
  messagingSenderId: "stuff",
  appId: "stuff",
  measurementId: "stuff"
}

firebase.initializeApp(firebaseConfig)
let db = firebase.firestore()
export default db`

I actually get one error when I view the console:
**Uncaught (in promise) TypeError: Cannot read property 'initializeApp' of undefined**

In the terminal window (I'm on a Mac) I get this error at the same time:
**export 'default' (imported as 'firebase') was not found in 'firebase/app'** 
   
I'm actually not sure which error I should be concentrating on.  I've been spending about 3 hours on Google trying to figure out why it's not working.  I am pretty new at Quasar and JavaScript and I'm still learning, so please be patient.  I've tried a number of different things and I can't get anything to work.

I appreciate any help

**Comment by ontwikkelfabriek** [comment]
@CynthiaBlue 

I think  the Firebase API changed a lot lately. So if you're not using the  same Firebase npm libs `versions` like Danny used, then it will probably not work. If you're new in JS and Quasar I would not go mess around with Firebase btw....Learn JS first then Vue then Quasar then ...

**Reply by CynthiaBlue** [reply]
@ontwikkelfabriek 
Good to know... things change so fast.  What alternative to firebase would you suggest? I could look into MySQL and host it on my own server.  I need something cheap or free. The website will be pretty low volume but will need to store data in a database.  

Thanks for your reply.

**Reply by ontwikkelfabriek** [reply]
@CynthiaBlue 

You can still use Firebase in Quasar. You could use the source code of Danny's Smackchat (Quasar, firebase, and more) as an example:
https://www.dconnell.co.uk/#/resource/smackchat-youtube-source-code

I can also recommend Strapi:
https://strapi.io

**Reply by CynthiaBlue** [reply]
@ontwikkelfabriek thanks for the info.  I am checking out the SmackChat app by Danny (love his videos).  Looks like he uses the Firebase Realtime database for that though, not the Cloud Firestore.  I'm wondering how difficult it would be to adapt it to Cloud Firestore.

**Reply by peppermintbunny** [reply]
I've only used Firestore so I cant help on this.

**Comment by peppermintbunny** [comment]
I had similar problems with Firebase on my Mac. 
Check that you are using Firestore version 8 and not Firestore version 9.  
Also, add "eslint" to your project.

**Reply by CynthiaBlue** [reply]
I do have Firebase V9.16.6 installed on my Mac.  I can try to install the v8 instead, and see if that will help.

**Reply by peppermintbunny** [reply]
I suggest you also:
 
1:  use yarn instead of npm (this is recommended)
2.  here is a command to downgrade your firebase:    yarn add firebase@^8.10.0
3.  when you have difficulties do these steps:
                a. delete   file “.quasar”
             b. delete folder  node-modules
              c. delete file “yearn.lock”
    then re-generate these 3 entities by typing   “yarn”
 
 
repeat step 3 whenever you have problems
 
 
Firebase works like a dream with quasar 2
